docs/_source/guides/llms/practical_guides/fine_tune.md

```diff
@@ -1,24 +1,305 @@
+|:--------------------------------|:-----|:-------|:----------|:-------|:------|:-------------|:-----|
+| Text Classification             |      | ✔️      | ✔️         | ✔️      | ✔️     | ✔️            | ✔️    |
+| Supervised Fine-tuning          | ✔️    |        |           |        |       |              |      |
+| Reward Modeling                 | ✔️    |        |           |        |       |              |      |
+| Proximal Policy Optimization    | ✔️    |        |           |        |       |              |      |
+| Direct Performance Optimization | ✔️    |        |           |        |       |              |      |
+
+
+```{note}
+We also offer support for  Token Classification using our `TokenClassifcationDataset` but this is shown in [a section](/guides/train_a_model) about our older dataset-types.
+```
+
+#### Training Configs
+
+The trainer also has an `ArgillaTrainer.update_config()` method, which maps a dict with `**kwargs` to the respective framework. So, these can be derived from the underlying framework that was used to initialize the trainer. Underneath, you can find an overview of these variables for the supported frameworks.
+
+```{note}
+Note that you don't need to pass all of them directly and that the values below are their default configurations.
+```
+
+```{include} /_common/tabs/train_update_config.md
+```
+
+### The `TaskMapping`
+
+A `TaskMapping` is used to define how the data should be processed and formatted according to the associated task and framework. Each task has its own `TaskMapping.for_*`-classmethod and the data formatting can always be defined using a custom `formatting_func`. However, simpler tasks like Text Classification can also be defined using default definitions. These directly use the fields and questions from the FeedbackDataset configuration to infer how to prepare the data. Underneath you can find an overview of the `TaskMapping`` requirements.
+
+| Method                             | Content          | Return formatting_func                                     | Default           |
+|:-----------------------------------|:-----------------|:-----------------------------------------------------------|:------------------|
+| for_text_classification            | `text-label`     | `Union[Tuple[str, str], Tuple[str, List[str]]]`            | ✔️                 |
+| for_supervised_fine_tuning         | `text`           | `str`                                                      | ✗                 |
+| for_reward_modelling               | `chosen-rejected`| `Tuple[str, str]`                                          | ✗                 |
+| for_proximal_policy_optimization   | `text`           | `Union[str, Iterator[str]]`                                | ✗                 |
+| for_direct_performance_optimization| `prompt-chosen-rejected`                 | `Tuple[str, str]`                                          | ✗                 |
```

Comment:
```suggestion
|:-----------------------------------|:-----------------|:-----------------------------------------------------------|:------------------|
| for_text_classification            | `text-label`     | `Union[Tuple[str, str], Tuple[str, List[str]]]`            | ✔️                 |
| for_supervised_fine_tuning         | `text`           | `Optional[Union[str, Iterator[str]]]`                                                      | ✗                 |
| for_reward_modelling               | `chosen-rejected`| `Optional[Union[Tuple[str, str], Iterator[Tuple[str, str]]]]`                                          | ✗                 |
| for_proximal_policy_optimization   | `text`           | `Optional[Union[str, Iterator[str]]]`                                | ✗                 |
| for_direct_performance_optimization| `prompt-chosen-rejected`                 | `Optional[Union[Tuple[str, str, str], Iterator[Tuple[str, str, str]]]]`                                          | ✗                 |
```

They all accept iterators and `None`. Not sure about text classification though.